pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -267,11 +279,15 @@ func validatePipelineTaskName(ctx context.Context, prefix string, i int, t Pipel
 		if errSlice := validation.IsQualifiedName(t.TaskRef.Name); len(errSlice) != 0 {
 			return apis.ErrInvalidValue(strings.Join(errSlice, ","), fmt.Sprintf(prefix+"[%d].taskRef.name", i))
 		}
-		if _, ok := taskNames[t.Name]; ok {
-			return apis.ErrMultipleOneOf(fmt.Sprintf(prefix+"[%d].name", i))
-		}
-		taskNames[t.Name] = struct{}{}
 	}
+
+	if hasTaskRef && !isCustomTask && t.TaskRef.Name == "" {
+		return apis.ErrInvalidValue(t.TaskRef, "taskRef must specify name")
+	}
+	if isCustomTask && t.TaskRef.Kind == "" {
+		return apis.ErrInvalidValue(t.TaskRef, "custom task ref must specify apiVersion and kind")
```

Comment:
There are a bunch of things that aren't supported initially with custom tasks:
* retries
* pipelineresources
* workspaces

There should be validation if the pipeline task tries to use these.

Reply:
Added validation and tests, thanks for catching!